src/main/java/minijava/ir/optimize/ConstantFolder.java

```diff
@@ -177,12 +253,20 @@ public void visit(Or node) {
 
   @Override
   public void visit(Phi node) {
-    phiStrategy.visit(node);
+    TargetValue newValue =
+        seq(node.getPreds())
+            .map(this::getValue)
+            .reduce(TargetValue.getUnknown(), ConstantFolder::supremum);
+    updateValue(node, newValue);
```

Comment:
I don't understand why we suddenly don't need the different strategies anymore? Could you elaborate?